### ISSUE TITLE
@visibleForTesting error message on named constructors gives wrong constructor name

### ISSUE THREAD

**Issue by Hixie** [dart-lang/linter#1986]
```dart
// main.dart
import 'inner.dart';

void main() {
  A.b();
}
```

```dart
// inner.dart
import 'package:meta/meta.dart';

class A {
  @visibleForTesting
  A.b();
}
```
->
```
   info • The member 'A' can only be used within 'inner.dart' or a test • main.dart:4:3 • invalid_use_of_visible_for_testing_member
```

It should say `A.b`, not `A`, in the error message.

**Comment by pq** [comment]
Good catch.  It looks like we've been mis-reporting diagnostics for named constructors generally.  Analyzer fix on the way... 👍

**Comment by pq** [comment]
Fixed in the SDK w/ https://github.com/dart-lang/sdk/commit/8ebe354ecc2c1e9c32cdd84053e05e7b5fc0ed74.
